app/services/foreman/plugin.rb

```diff
@@ -234,15 +238,34 @@ def permission(name, hash, options = {})
 
     # Add a new role if it doesn't exist
     def role(name, permissions)
-      @default_roles[name] = permissions
+      default_roles[name] = permissions
       return false if pending_migrations || Rails.env.test?
 
       Role.transaction do
         role = Role.where(:name => name).first_or_create
```

Comment:
now when we lock shipped roles, should role created by plugin be also locked? and in case we find existing (from previous installation, we should probably make sure we lock it from now on)